test/utils/wait_for.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:meilisearch/meilisearch.dart';
+import 'package:meilisearch/src/task.dart';
+
+extension TaskWaiter on Task {
+  Future<Task> waitFor({
+    required MeiliSearchClient client,
+    Duration timeout = const Duration(seconds: 5),
+    Duration interval = const Duration(milliseconds: 50),
+  }) async {
+    var endingTime = DateTime.now().add(timeout);
+
+    while (DateTime.now().isBefore(endingTime)) {
+      var task = await client.getTask(this.uid!);
```

Comment:
If the this is provided by the `addDocuments` method, should it not be `taskUid` ?

Reply:
In the Dart SDK, you just have `uid`.